opentrons-ai-client/src/molecules/ModulesButtonGroup/index.tsx

```diff
@@ -32,10 +34,25 @@ export function ModulesButtonGroup({
                   key={module.type}
                   iconName="plus"
                   onClick={() => {
-                    if (modulesWatch.some(m => m.type === module.type)) {
-                      return
+                    const isTempModule = module.type === TEMPERATURE_MODULE_TYPE
+                    const tempModuleCount = modulesWatch.filter(
+                      m => m.type === TEMPERATURE_MODULE_TYPE
+                    ).length
+
+                    if (isTempModule) {
+                      if (tempModuleCount >= 2) {
+                        return
+                      }
+                    } else {
+                      if (modulesWatch.some(m => m.type === module.type)) {
+                        return
+                      }
+                    }
```

Comment:
let's avoid using the nested if-statements as much as possible.

```suggestion
if (
  (isTempModule && tempModuleCount >= 2) ||
  (!isTempModule && modulesWatch.some(m => m.type === module.type))
) {
  return;
}
```

Reply:
Fixed!